### PR TITLE
TSV support

### LINKED ISSUE #4746: TSV read/write
At the time this issue is being opened, Zed it at commit b8e1b18.

A user asked in a [community Slack thread](https://brimdata.slack.com/archives/CTSMAK6G7/p1691681741869439):

> I would like for the `zq` command support the TSV-format (Tab-Separated-Values).
> This would make it possible to pipe the output from more programs directly to `zq` for post-processing. For instance if I select a range of cells in Excel and paste it into the terminal I get TSV-data from Excel. Processed with something like `pbpaste | zq -Z -` on a Mac.  Another program that I often use is [VisiData](https://www.visidata.org/) that also outputs TSV by default in pipelines.

We reviewed this one as a team and there was agreement that, particularly given the prevalence of these Excel (and Google Sheets) use cases, it would indeed be useful for us to have direct support for this, e.g.:

1. Explicit `-i tsv` for TSV input
2. Supporting TSV via auto-detect, if feasible
3. `-f tsv` for TSV output

While waiting for this to be implemented, it should be noted that there is already TSV support on the input side, though it's not the easiest to use. The `-csv.delim` option added in #4325 allows the specification of any delimiter when reading via `-i csv`. Unfortunately, it can be kind of a pain to get the tab character past the shell. For instance, given this input data [wine.tsv.gz](https://github.com/brimdata/zed/files/12315989/wine.tsv.gz):

```
$ gzcat wine.tsv.gz 
#	Year	Winery
1	2019	Brutocao
1	2021	Cabana
1	2020	Castoro
```

With my Bash shell, after I type the first `"`, I hit Ctrl-v before hitting Tab and then type the second `"`, and this gets the tab through to `zq` (though unfortunately this doesn't survive cut & paste into & out of the GitHub Issue, so users will need to manually complete the steps in their own shells to see it working).

```
$ zq -version
Version: v1.9.0-7-gb8e1b188

$ zq -i csv -csv.delim "	" wine.tsv.gz 
{"#":1.,Year:2019.,Winery:"Brutocao"}
{"#":1.,Year:2021.,Winery:"Cabana"}
{"#":1.,Year:2020.,Winery:"Castoro"}
```

### PR DESCRIPTION
**Pull request by mattnibs**
Add tsv for support for both input and output. Also support auto-detection of tsv input.

Closes #4746

### CODE REVIEW

**Comment by philrz** [comment]
I just tested out this branch at commit 86b9fdb and it worked great for me with the `pbpaste | zq -Z -` Google Sheets use case described in #4746. So it's a functional 👍 for me!

File: zio/anyio/lookup.go [hunk 1]
```diff
@@ -41,6 +41,9 @@ func lookupReader(zctx *zed.Context, r io.Reader, demandOut demand.Demand, opts
 			return nil, err
 		}
 		return zio.NopReadCloser(zr), nil
+	case "tsv":
+		opts.CSV.Delim = '\t'
+		return zio.NopReadCloser(csvio.NewReader(zctx, r, opts.CSV)), nil
```
Comment (nwt) [inline zio/anyio/lookup.go:46]: Put this between Parquet and VNG.

File: zio/anyio/reader.go [hunk 2]
```diff
@@ -118,16 +118,14 @@ func NewReaderWithOpts(zctx *zed.Context, r io.Reader, demandOut demand.Demand,
...
-		csvErr = fmt.Errorf("csv: line 1: %w", err)
-	} else if !strings.Contains(s, ",") {
-		csvErr = errors.New("csv: line 1: no comma found")
-	} else {
-		track.Reset()
-		csvErr = match(csvio.NewReader(zed.NewContext(), track, opts.CSV), "csv", 1)
-		if csvErr == nil {
-			return zio.NopReadCloser(csvio.NewReader(zctx, track.Reader(), opts.CSV)), nil
-		}
+	if csvErr = isCSVStream(track, ',', "csv"); csvErr == nil {
```
Comment (nwt) [inline zio/anyio/reader.go:121]:
```suggestion
	csvErr := isCSVStream(track, ',', "csv")
	if csvErr == nil {
```

File: zio/anyio/reader.go [hunk 3]
```diff
@@ -118,16 +118,14 @@ func NewReaderWithOpts(zctx *zed.Context, r io.Reader, demandOut demand.Demand,
...
+		return zio.NopReadCloser(csvio.NewReader(zctx, track.Reader(), csvio.ReaderOpts{Delim: ','})), nil
+	}
+	track.Reset()
+
+	var tsvErr error
+	if tsvErr = isCSVStream(track, '\t', "tsv"); tsvErr == nil {
```
Comment (nwt) [inline zio/anyio/reader.go:127]:
```suggestion
	tsvErr := isCSVStream(track, '\t', "tsv")
	if tsvErr == nil {
```

File: zio/anyio/reader.go [hunk 4]
```diff
@@ -176,6 +175,16 @@ func isArrowStream(track *Track) error {
...
 }
 
+func isCSVStream(track *Track, delim rune, name string) error {
+	if s, err := bufio.NewReader(track).ReadString('\n'); err != nil {
+		return fmt.Errorf("%s: line 1: %w", name, err)
+	} else if !strings.Contains(s, string(delim)) {
+		return fmt.Errorf("%s: line 1: delimiter %q not found", name, delim)
+	}
+	track.Reset()
+	return match(csvio.NewReader(zed.NewContext(), track, csvio.ReaderOpts{Delim: delim}), "tsv", 1)
```
Comment (nwt) [inline zio/anyio/reader.go:185]:
```suggestion
	return match(csvio.NewReader(zed.NewContext(), track, csvio.ReaderOpts{Delim: delim}), name, 1)
```

**Review by nwt — COMMENTED** [review]
1. Add `MediaTypeTSV = "application/tab-separated-values"`, wire it up, add tests in services/ztests/, add it to docs/lake/api.md.
2. Add TSV to the Input Formats section in docs/commands/zq.md.

**Review by nwt — APPROVED** [review]

**Pull request merged**